cartography/intel/azure/compute.py

```diff
@@ -0,0 +1,227 @@
+import logging
+from typing import Dict
+from typing import List
+
+import neo4j
+from azure.mgmt.compute import ComputeManagementClient
+
+from .credentials import Credentials
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+def get_client(credentials: Credentials, subscription_id: str) -> ComputeManagementClient:
+    client = ComputeManagementClient(credentials, subscription_id)
+    return client
+
+
+def get_vm_list(credentials: Credentials, subscription_id: str) -> List[Dict]:
+    try:
+        client = get_client(credentials, subscription_id)
+        vm_list = list(map(lambda x: x.as_dict(), client.virtual_machines.list_all()))
+
+        for vm in vm_list:
+            x = vm['id'].split('/')
+            vm['resource_group'] = x[x.index('resourceGroups') + 1]
+
+        return vm_list
+
+    except Exception as e:
```

Comment:
Can we scope this to a specific exception?

Reply:
updated code to use specific exception